cpp/src/arrow/util/decimal-test.cc

```diff
@@ -380,4 +380,16 @@ TEST(Decimal128Test, TestNoDecimalPointExponential) {
   ASSERT_EQ(0, scale);
 }
 
+TEST(Decimal128Test, TestFromBigEndian) {
+  Decimal128 value(1);
```

Comment:
Ideally you should also test non-trivial values (not one which only has only one bit set), and also negative values.